### ISSUE TITLE
Allow user submission to the AccurateRip database

### ISSUE THREAD

**Issue by BoBeR182** [whipper-team/whipper#36]
``` bash
AccurateRip URL http://www.accuraterip.com/accuraterip/a/6/8/dBAR-012-000e686a-008a4eb8-84081a0c.bin
Album not found in AccurateRip database
Track  1: rip NOT accurate (not found)             [e5f61ed9], DB [notfound]
Track  2: rip NOT accurate (not found)             [bcb9ac0a], DB [notfound]
Track  3: rip NOT accurate (not found)             [db4767bf], DB [notfound]
Track  4: rip NOT accurate (not found)             [fe83e31d], DB [notfound]
Track  5: rip NOT accurate (not found)             [f44f789e], DB [notfound]
Track  6: rip NOT accurate (not found)             [cbf7fbcc], DB [notfound]
Track  7: rip NOT accurate (not found)             [c7a348d6], DB [notfound]
Track  8: rip NOT accurate (not found)             [f412d33b], DB [notfound]
Track  9: rip NOT accurate (not found)             [6013ff9a], DB [notfound]
Track 10: rip NOT accurate (not found)             [3d75a847], DB [notfound]
Track 11: rip NOT accurate (not found)             [dbd7661f], DB [notfound]
Track 12: rip NOT accurate (not found)             [89807686], DB [notfound]
```

Can we add rips to the database to help it grow?

**Comment by JoeLametta** [comment]
> Can we add rips to the database to help it grow?

Not really: if I recall correctly the only softwares allowed to submit records to AccurateRip are dbPoweramp (Windows version, standalone) & Exact Audio Copy (EAC) with the AccurateRip plugin installed.

[Here](http://wiki.hydrogenaud.io/index.php?title=AccurateRip) it is explained with more details:
> The database is maintained by Illustrate Ltd., the company run by "Spoon", primary developer of dBpoweramp. All of the data is submitted by the Windows-based ripping programs dBpoweramp and Exact Audio Copy via a DLL licensed from Illustrate.[[1]](http://wiki.hydrogenaud.io/index.php?title=AccurateRip#cite_ref-1) Other rippers and tools, such as CUETools and Mac OS-based rippers, can receive info from the database in order to look up drive offsets or check rip quality, but they cannot submit their own rip results to it. 

What can be done here is improving whipper's AccurateRip support putting it on par with EAC's one. That's explained in issue #18.

**Comment by MerlijnWajer** [comment]
Technically it is not impossible to do this, but will require some reverse engineering of the AccurateRip code/binary/dll. Or just sniff the protocol. Whether EAC/dbP will like it is another question.

**Comment by BoBeR182** [comment]
Has anyone offered to get a API key or the DLL ported to linux?

**Comment by JoeLametta** [comment]
> Technically it is not impossible to do this, but will require some reverse engineering of the AccurateRip code/binary/dll. Or just sniff the protocol. Whether EAC/dbP will like it is another question.

Right but I personally don't want to override Spoon's will and I'm also not capable of doing that (of course anybody is free to work in a different direction).

> Has anyone offered to get a API key or the DLL ported to linux?

I don't know but don't think so...

Maybe, if this feature is really desired, will be wiser to focus energies on issue #15 (which shouldn't prohibit submissions coming from other rippers). Anyway this is just an idea: no work will be done in this regard in the foreseeable future.

**Comment by MerlijnWajer** [comment]
I don't believe that the dll is very complicated. I agree that this is not a priority, but I also think it's probably not too hard to reimplement this functionality.

**Comment by Freso** [comment]
If this is "wontfix", why is it still open? Either the "wontfix" label should be removed, or this should be closed… ?

**Comment by JoeLametta** [comment]
> If this is "wontfix", why is it still open? Either the "wontfix" label should be removed, or this should be closed… ?

Yeah, I'm going to close it now (no activity since 2016/09).

**Comment by MerlijnWajer** [comment]
Can't we keep this open as a wishlist-feature? Seems like this will pop up again at some point.
